nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -182,7 +195,9 @@ def main():
         # be there on newly installed systems, so let's generate one so that
         # bootctl can find it and we can also pass it to write_entry() later.
         cmd = ["@systemd@/bin/systemd-machine-id-setup", "--print"]
-        machine_id = subprocess.check_output(cmd).rstrip()
+        machine_id = subprocess.run(
+          cmd, text=True, check=True, stdout=subprocess.PIPE
+        ).stdout.rstrip()
```

Comment:
Why not the following?

```suggestion
        machine_id = subprocess.check_output(cmd, text=True).rstrip()
```

So far I'm not aware that `check_output` is deprecated, did I miss something?

Reply:
Not really deprecated (they're not going to be removed), but `run` is the recommended way in almost all cases: https://docs.python.org/3.8/library/subprocess.html#using-the-subprocess-module

BTW, the documentation of the older APIs are here: https://docs.python.org/3.8/library/subprocess.html#older-high-level-api

I particularly find `run` much more readable. Instead of trying to remember what function to use in each case, now you simply need to enable/disable the flags until you have the desired behavior.

Reply:
I did not know that check_output has also a `text` argument. I mainly changed it because before it was returning bytes before which breaks the type check. However people in a few years will probably be more familiar with run than all the other functions.